### PR TITLE
adding key -F to pg_dump

### PR DESCRIPTION
**Pull request by Fenimorkin**
Added key unnecessary formate for dump()

-F format
--format=format
    Selects the format of the output. format can be one of the following:
    p
    plain
        Output a plain-text SQL script file (the default). 
    c
    custom
        Output a custom-format archive suitable for input into pg_restore. Together with the directory output format, this is the most flexible output format in that it allows manual selection and reordering of archived items during restore. This format is also compressed by default. 
    d
    directory

        Output a directory-format archive suitable for input into pg_restore. This will create a directory with one file for each table and blob being dumped, plus a so-called Table of Contents file describing the dumped objects in a machine-readable format that pg_restore can read. A directory format archive can be manipulated with standard Unix tools; for example, files in an uncompressed archive can be compressed with the gzip tool. This format is compressed by default and also supports parallel dumps. 
    t
    tar
        Output a tar-format archive suitable for input into pg_restore. The tar format is compatible with the directory format: extracting a tar-format archive produces a valid directory-format archive. However, the tar format does not support compression. Also, when using tar format the relative order of table data items cannot be changed during restore.

### CODE REVIEW

**Comment by stalkerg** [comment]
You should fix code style for this pull request.

**Comment by ildus** [comment]
Tests should based on unittest library.

File: testgres/node.py [hunk 1]
```diff
@@ -583,8 +583,8 @@ def cleanup(self, max_attempts=3):
     def psql(self,
              dbname,
              query=None,
-             filename=None,
```
Comment (funbringer) [inline testgres/node.py]: Arguments should not be moved around.

File: testgres/node.py [hunk 2]
```diff
@@ -658,7 +658,7 @@ def safe_psql(self, dbname, query, username=None, input=None):
 
         return out
 
-    def dump(self, dbname, username=None, filename=None):
+    def dump(self, dbname, username=None, filename=None, formate=None):
```
Comment (funbringer) [inline testgres/node.py]: Probably should replace `formate` with `format` (I know, it's a function name, but who cares).

File: testgres/node.py [hunk 3]
```diff
@@ -667,13 +667,15 @@ def dump(self, dbname, username=None, filename=None):
...
             filename: output file.
+            formate: format argument p/c/d/t
 
         Returns:
             Path to a file containing dump.
         """
 
         # Set default arguments
         username = username or _default_username()
+        formate = formate or "p"
```
Comment (funbringer) [inline testgres/node.py]: "p" should be replaced with a const defined in consts.py

File: tests/test_pgdump.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    dump = node1.dump('postgres')
+    with get_new_node('node2') as node2:
+        node2.init().start().restore('postgres', dump)
+        res = node2.execute('postgres','select * from test order by val asc')
+    os.remove(dump)
+
+# 1) make dump to new place
+#     pg_dump mydb > db.sql 
+# 2) make dump to non default formate 
+#     2.5) Чтобы сформировать выгрузку в формате plain:
```
Comment (funbringer) [inline tests/test_pgdump.py]: Russian comments is a big no-no.

**Comment by stalkerg** [comment]
I think we should drop this PR.

**Comment by funbringer** [comment]
@stalkerg I believe we should give @Fenimorkin a chance to finish his work.

File: testgres/node.py (hunk 2 above)
Comment (Fenimorkin) [reply testgres/node.py]: done

File: testgres/node.py (hunk 3 above)
Comment (Fenimorkin) [reply testgres/node.py]: done

**Pull request merged**